src/app/pages/test-report/test-report.effects.ts

```diff
@@ -0,0 +1,149 @@
+          }),
+        );
+    }),
+  ));
+
+  persistTestReport$ = this.actions$.pipe(
+    ofType(
+      drivingFaultsActions.AddDrivingFault,
+      drivingFaultsActions.RemoveDrivingFault,
+      seriousFaultsActions.RemoveSeriousFault,
+      seriousFaultsActions.AddSeriousFault,
+      dangerousFaultsActions.AddDangerousFault,
+      dangerousFaultsActions.RemoveDangerousFault,
+      manoeuvresActions.RecordManoeuvresSelection,
+      manoeuvresActions.AddManoeuvreDrivingFault,
+      manoeuvresActions.AddManoeuvreSeriousFault,
+      manoeuvresActions.AddManoeuvreDangerousFault,
+      manoeuvresActions.RemoveManoeuvreFault,
+      manoeuvresActions.RecordManoeuvresDeselection,
+      uncoupleRecoupleActions.ToggleUncoupleRecouple,
+      uncoupleRecoupleActions.UncoupleRecoupleAddDrivingFault,
+      uncoupleRecoupleActions.UncoupleRecoupleAddSeriousFault,
+      uncoupleRecoupleActions.UncoupleRecoupleAddDangerousFault,
+      uncoupleRecoupleActions.UncoupleRecoupleRemoveFault,
+      controlledStopActions.ToggleControlledStop,
+      controlledStopActions.ControlledStopAddDrivingFault,
+      controlledStopActions.ControlledStopAddSeriousFault,
+      controlledStopActions.ControlledStopAddDangerousFault,
+      controlledStopActions.ControlledStopRemoveFault,
+      vehicleChecksActions.ShowMeQuestionPassed,
+      vehicleChecksActions.ShowMeQuestionDrivingFault,
+      vehicleChecksActions.ShowMeQuestionSeriousFault,
+      vehicleChecksActions.ShowMeQuestionDangerousFault,
+      vehicleChecksActions.ShowMeQuestionRemoveFault,
+      ecoActions.ToggleEco,
+      ecoActions.ToggleControlEco,
+      ecoActions.TogglePlanningEco,
+      etaActions.ToggleETA,
+      testRequirementsActions.ToggleLegalRequirement,
+      avoidanceActions.AddAvoidanceSeriousFault,
+      avoidanceActions.RemoveAvoidanceSeriousFault,
+      avoidanceActions.RecordAvoidanceFirstAttempt,
+      avoidanceActions.RecordAvoidanceSecondAttempt,
+      emergencyStopActions.AddEmergencyStopSeriousFault,
+      emergencyStopActions.RemoveEmergencyStopSeriousFault,
+      emergencyStopActions.RecordEmergencyStopFirstAttempt,
+      emergencyStopActions.RecordEmergencyStopSecondAttempt,
+      singleFaultCompetenciesActions.SetSingleFaultCompetencyOutcome,
+      singleFaultCompetenciesActions.RemoveSingleFaultCompetencyOutcome,
+      singleFaultCompetenciesActions.RemoveSingleSeriousFaultCompetencyOutcome,
+      singleFaultCompetenciesActions.RemoveSingleDangerousFaultCompetencyOutcome,
+      highwayCodeActions.HighwayCodeSafetyAddDrivingFault,
+      highwayCodeActions.HighwayCodeSafetyAddSeriousFault,
+      highwayCodeActions.HighwayCodeSafetyRemoveFault,
+      highwayCodeActions.ToggleHighwayCodeSafety,
+    ),
+    concatMap((action) => of(action).pipe(
+      withLatestFrom(
+        this.store$.pipe(
+          select(getTests),
+          map(isTestReportPracticeTest),
+        ),
+      ),
+    )),
+    // eslint-disable-next-line no-unused-vars
```

Comment:
unless im mistaken if there is no _ in the filter below eslint shouldnt pick it up as an unused var and this could be removed

Reply:
removed